src/sentry/api/serializers/models/event.py

```diff
@@ -92,15 +100,23 @@ def _get_tags_with_meta(self, event):
         tags = sorted(
             [
                 {
-                    'key': kv[0].split('sentry:', 1)[-1],
-                    'value': kv[1],
-                    '_meta': meta.get(kv[0]) or get_path(meta, six.text_type(i), '1') or None,
+                    'key': key.split('sentry:', 1)[-1],
+                    'value': value,
+                    '_meta': meta.get(key) or get_path(meta, six.text_type(i), '1') or None,
                 }
-                for i, kv in enumerate(event.data.get('tags') or ())
-                if kv is not None and kv[0] is not None and kv[1] is not None],
+                # TODO this should be using the Event.tags getter
+                for i, (key, value) in enumerate(event.data.get('tags') or ())
```

Comment:
If someone ingests `{"tags": [1, 2, ["foo", "bar"]]}` then `{"tags": [null, null, ["foo", "bar"]}` is stored which the old code handled and the new does not.

Reply:
Wouldn't the original code here choke on that input as well? 
```
In [1]: kv = 1
In [2]: kv is not None and kv[0] is not None and kv[1] is not None
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-21c0faadfa28> in <module>()
----> 1 kv is not None and kv[0] is not None and kv[1] is not None

TypeError: 'int' object has no attribute '__getitem__'
```

Reply:
The normalizer coerces an item of bad type (in this case `1`) into `null` and attaches an error into the metadata out of bounds. The `kv is not None` caught that null case. You would never see an int there.

Reply:
```python
>>> e = EventManager({'tags': [1, 2, ['test', 'it']]})
>>> e.normalize()
>>> e.get_data()['tags']
[None, None, [u'test', u'it']]
```

Reply:
Went back to old logic.